isis/tests/Fixtures.cpp

```diff
@@ -253,7 +255,110 @@ namespace Isis {
+    isdFile3 = new FileName("data/apolloNetwork/Sub4-AS15-M-0585_msk.isd");
+    isdFile4 = new FileName("data/apolloNetwork/Sub4-AS15-M-0586_msk.isd");
+    isdFile5 = new FileName("data/apolloNetwork/Sub4-AS15-M-0587_msk.isd");
+    isdFile6 = new FileName("data/apolloNetwork/Sub4-AS15-M-1423.isd");
+    isdFile7 = new FileName("data/apolloNetwork/Sub4-AS15-M-1537.isd");
+
+    label1 = new FileName("data/apolloNetwork/Sub4-AS15-M-0583_msk.pvl");
+    label2 = new FileName("data/apolloNetwork/Sub4-AS15-M-0584_msk.pvl");
+    label3 = new FileName("data/apolloNetwork/Sub4-AS15-M-0585_msk.pvl");
+    label4 = new FileName("data/apolloNetwork/Sub4-AS15-M-0586_msk.pvl");
+    label5 = new FileName("data/apolloNetwork/Sub4-AS15-M-0587_msk.pvl");
+    label6 = new FileName("data/apolloNetwork/Sub4-AS15-M-1423.pvl");
+    label7 = new FileName("data/apolloNetwork/Sub4-AS15-M-1537.pvl");
+
+    cube1 = new Cube();
+    cube1->fromIsd(tempDir.path() + "/cube1.cub", *label1, *isdFile1, "rw");
+
+    cube2 = new Cube();
+    cube2->fromIsd(tempDir.path() + "/cube2.cub", *label2, *isdFile2, "rw");
+
+    cube3 = new Cube();
+    cube3->fromIsd(tempDir.path() + "/cube3.cub", *label3, *isdFile3, "rw");
+
+    cube4 = new Cube();
+    cube4->fromIsd(tempDir.path() + "/cube4.cub", *label4, *isdFile4, "rw");
+
+    cube5 = new Cube();
+    cube5->fromIsd(tempDir.path() + "/cube5.cub", *label5, *isdFile5, "rw");
+
+    cube6 = new Cube();
+    cube6->fromIsd(tempDir.path() + "/cube6.cub", *label6, *isdFile6, "rw");
+
+    cube7 = new Cube();
+    cube7->fromIsd(tempDir.path() + "/cube7.cub", *label7, *isdFile7, "rw");
+
+    cubeList = new FileList();
+    cubeList->append(cube1->fileName());
+    cubeList->append(cube2->fileName());
+    cubeList->append(cube3->fileName());
+    cubeList->append(cube4->fileName());
+    cubeList->append(cube5->fileName());
+    cubeList->append(cube6->fileName());
+    cubeList->append(cube7->fileName());
+
+    cubeListFile = tempDir.path() + "/cubes.lis";
+    cubeList->write(cubeListFile);
+
+    ControlNet inputNet("data/apolloNetwork/apolloNet.pvl");
+    inputNet.Write(tempDir.path() + "/apolloNet.net");
+  }
+
+  void ApolloNetwork::TearDown() {
+    if (cube1->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube2->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube3->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube4->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube5->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube6->isOpen()) {
+      cube1->close();
+    }
+
+    if (cube7->isOpen()) {
+      cube1->close();
+    }
+
+    delete isdFile1;
+    delete isdFile2;
+    delete isdFile3;
+    delete isdFile4;
+    delete isdFile5;
+    delete isdFile6;
+    delete isdFile7;
+
+    delete cube1;
+    delete cube2;
+    delete cube3;
+    delete cube4;
+    delete cube5;
+    delete cube6;
+    delete cube7;
+
+    delete cubeList;
```

Comment:
We really should make these fixtures more careful about just raw deleting things. Either we should move to smart pointers or we should wrap our deletes in if checks.

Reply:
I feel like smart pointers is a better idea, but I just updated to wrap in "if checks" for now.